app/components/ChangePassphraseModal/index.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import React from "react";
+import { autobind } from "core-decorators";
+import Modal from "./Modal";
+
+@autobind
+class ChangePassphraseModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this.getInitialState();
+  }
+
+  getInitialState() {
+    return {
+      privpass: null,
+      oldPrivPass: null,
+      confirmPrivPass: null,
+      privPassError: null,
+      oldPrivPassError: null,
+      confirmPrivPassError: null
+    };
+  }
+
+  render() {
+    const { hidden, heading, description } = this.props;
+    const {
+      privpass,
+      oldPrivPass,
+      confirmPrivPass,
+      privPassError,
+      oldPrivPassError,
+      confirmPrivPassError
+    } = this.state;
+    const {
+      updateOldPrivatePassphrase,
+      updatePrivatePassphrase,
+      updateConfirmPrivatePassphrase,
+      submitPassphrase,
+      cancelPassphrase
+    } = this;
+
+    return (
+      <Modal
+        {...{
+          hidden,
+          heading,
+          description,
+          privpass,
+          oldPrivPass,
+          confirmPrivPass,
+          privPassError,
+          oldPrivPassError,
+          confirmPrivPassError,
+          updateOldPrivatePassphrase,
+          updatePrivatePassphrase,
+          updateConfirmPrivatePassphrase,
+          submitPassphrase,
+          cancelPassphrase
+        }}
+      />
+    );
+  }
+
+  resetState() {
+    this.state = this.getInitialState(); // clear password data from memory
+    document.getElementById("passphrase").value = "";
```

Comment:
I'll test with and without these clears, if functionally it is the same then I'm cool with them going.